src/Hosting/ResponsivePageRouteModelConvention.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc.ApplicationModels;
+using Wangkanai.Detection.Models;
+
+namespace Wangkanai.Detection.Hosting
+{
+    internal class ResponsivePageRouteModelConvention : IPageRouteModelConvention
+    {
+        public void Apply(PageRouteModel model)
+        {
+            var fileNameWithoutExtension = model.ViewEnginePath.Split('/')[^1];
+            if (!fileNameWithoutExtension.Contains('.'))
+            {
+                // Normal page, nothing to do!
+                return;
+            }
+
+            // This just implements the 'suffix' strategy
+            var split = fileNameWithoutExtension.Split('.');
+            if (split.Length != 2)
+            {
+                throw new InvalidOperationException($"Page '{model.RelativePath}' does not follow the required format.");
+            }
+
+            var pageName = split[0];
+            var deviceName = split[1];
+
+            if (!Enum.TryParse<Device>(deviceName, ignoreCase: true, out var device))
+            {
+                throw new InvalidOperationException($"Device name could not be parsed for page '{model.RelativePath}'.");
+            }
+
+            // Since the page name has something like `.mobile` in it, the special cased rules for Index.cshtml
+            // don't apply. We know there's going to be one selector.
+            var selector = model.Selectors.Single(); 
```

Comment:
I will need to investigate and learn more about this `PageRouteModel` and `SelectorModel` later to have deeper understanding of it.